tests/jobserver/api/test_releases.py

```diff
@@ -82,140 +82,196 @@ def test_releasenotificationapicreate_with_failed_slack_update(
+    assert response["Location"] == f"http://testserver{release.get_api_url()}"
 
 
 @pytest.mark.django_db
-def test_upload_api_redirected():
-    org = OrgFactory()
-    project = ProjectFactory(org=org)
-    workspace = WorkspaceFactory(project=project)
-    backend = BackendFactory(auth_token="test")
-    files = {"file.txt": "test redirect"}
-    upload = ReleaseUploadFactory(files)
-
-    # release already exists
-    release = ReleaseFactory(workspace=workspace, backend=backend, files=files)
-    assert Release.objects.count() == 1
+def test_workspace_api_create_release_bad_json():
+    user = UserFactory()
+    workspace = WorkspaceFactory()
+    ProjectMembershipFactory(user=user, project=workspace.project)
+    BackendFactory(auth_token="test")
 
-    response = APIClient().put(
-        f"/api/v2/workspaces/{workspace.name}/releases/{upload.hash}",
-        content_type="application/octet-stream",
-        data=upload.zip.read(),
+    response = APIClient().post(
+        f"/api/v2/releases/workspace/{workspace.name}",
+        content_type="application/json",
+        data=json.dumps({}),
         HTTP_CONTENT_DISPOSITION="attachment; filename=release.zip",
         HTTP_AUTHORIZATION="test",
-        HTTP_BACKEND_USER="user",
+        HTTP_OS_USER=user.username,
     )
 
-    assert response.status_code == 303
-    assert Release.objects.count() == 1
+    assert response.status_code == 400
+
 
-    release.refresh_from_db()
-    assert response["Release-Id"] == release.id
-    assert response["Location"] == f"http://testserver{release.get_absolute_url()}"
+@pytest.mark.django_db
+def test_workspace_index_api_not_exists():
+    client = APIClient()
+    response = client.get("/api/v2/releases/workspace/badid")
+    assert response.status_code == 404
 
 
 @pytest.mark.django_db
-def test_release_index_api_not_exists():
+def test_workspace_index_api_anonymous():
+    workspace = WorkspaceFactory()
+    client = APIClient()
+    index_url = f"/api/v2/releases/workspace/{workspace.name}"
+
+    response = client.get(index_url)
+    assert response.status_code == 403
+
+
+@pytest.mark.django_db
+def test_workspace_index_api_logged_in_no_permission():
+    workspace = WorkspaceFactory()
+    client = APIClient()
+    user = UserFactory()
+    index_url = f"/api/v2/releases/workspace/{workspace.name}"
+
+    client.force_authenticate(user=user)
+    response = client.get(index_url)
+    assert response.status_code == 403
+
+
+@pytest.mark.django_db
+def test_workspace_index_api_have_permission():
+    workspace = WorkspaceFactory()
+    backend1 = BackendFactory(name="backend1")
+    backend2 = BackendFactory(name="backend2")
+    user = UserFactory()
     client = APIClient()
+    client.force_authenticate(user=user)
+    ProjectMembershipFactory(
+        user=user, project=workspace.project, roles=[ProjectCollaborator]
+    )
+
+    index_url = f"/api/v2/releases/workspace/{workspace.name}"
```

Comment:
Can we keep the URL that's under test next to where it's used?  (Don't mind if that's inline to the `.get()` or just before it!)